porting justin's edits into GH (#253)
jaybee84/ml-in-rd: content/05.model-complexity.md

## Reducing misinterpretation of model output with statistical techniques

The successful application of ML can be improved by meeting certain conditions.

> First, the dataset contains an equal number of samples for all categories (no “class imbalance”).

Author: Justin: Don’t agree with the characterization. Balance is not required; more important is sufficient representation of each category, capturing the biological variability for that class

Author: ```suggestion
First, if all categories in a dataset have comparable variability, the dataset contains an equal number of samples for all categories (no “class imbalance”).
```

Reviewer: i think the takeaway that we should include here is different than what we have currently written. IMO class imbalance is a problem in rare disease less so because of the training, and more so in the testing/evaluation. That is, if you have a classifier to identify ultra rare disease vs non ultra rare disease people, it can be very accurate (e.g. >0.99) on an imbalanced test if it just guesses that everyone does not have the disease. 

I think the takeaway here should be (1) you need to have a large enough number of rare disease observations to capture the variation in features associated with that rare disease and (2) that evaluation metrics need to be carefully considered depending on the question being asked. The takeaways aren't really rare disease specific, but the fact that class imbalance will exist simply by virtue of looking at rare disease is somewhat rare disease specific.  

what do you all think?

Author: Agree with both (1) and (2). I think we make the case for (1) in the heterogeneity section. We have not explicitly stated (2) but the best place for that may be the outlook section.

This section mostly delves into the statistical assumptions that an ML method has, given that originally they are developed for large datasets. This section later goes into examples of statistical techniques like ensemble methods and regularization which help generate synthetic data points and reduce feature space to suit the sample space better respectively.

Reviewer 2: Agree with both (1) and (2) as well. However, I don't think (1) is sufficiently captured in the heterogeneity section as currently written. (2) should probably at least get mentioned in Box 2 in addition to the outlook section.

I will have a PR to this branch that addresses the latter. This sentence should get re-written to address (1) in my opinion.

Reviewer 2: Maybe?

```suggestion
First, the dataset contains sufficient representation from each class such that relevant variability from that class is captured.
```

And then another sentence that ties this to class imbalance.

I am not successfully reducing words here, but do we absolutely need to talk about class imbalance in this paragraph? Would it be sufficient to talk about it later in the context of what is now line 24?

Reviewer: I am a bit late here but I think the current version looks pretty good! I will be interested to see what Justin thinks.

Random forests use sampling with replacement based techniques to form a consensus about the important predictive features identified by the decision trees (e.g., Box 1c). [@https://doi.org/10.1023/A:1010933404324; @doi:10.1186/1472-6947-13-134]
Additional approaches like combining random forests with sampling without replacement can generate confidence intervals for the model predictions (for applications like Box 1d) by mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].
Resampling approaches are most helpful in constructing confidence intervals for algorithms that generate the same outcome every time they are run (i.e., deterministic models).
For decision trees that choose features at random for selecting a path to the outcome (i.e., are non-deterministic), resampling approaches can be helpful in estimating the reproducibility of the model.

In situations where decision tree-based ensemble methods fail when applied to rare disease datasets, cascade learning is a viable alternative. [@pmc:PMC6371307]

> In cascade learning, multiple methods leveraging distinct underlying assumptions are used to identify stable patterns in the dataset [@https://doi.org/10.1109/CVPR.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680].

Author: Justin: What type of assumptions?

Author: This statement was to highlight the difference between random forest and cascade learning, another ensemble method that uses different ML methods in tandem. The ML methods in cascade learning are a variety of techniques each with its own assumptions, while random forest is all decision trees. 

I think this has become obscure through our passage of edits.

Author: ```suggestion
In cascade learning, multiple methods leveraging distinct statistical techniques are used to identify stable patterns in the dataset [@https://doi.org/10.1109/CVPR.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680].
```

The presence of label-noise and sparsity in the data can lead to poor generalizability or overfitting, meaning that the models show high prediction accuracy on the training data but low prediction accuracy on new evaluation data.
Overfit models tend to rely on patterns that are unique to the training data, such as the clinical coding practices at a hospital, and not generalize to new data such as data collected at different hospitals. [@isbn:0262035618; @pmc:PMC8238368]
Regularization approaches can help mitigate these scenarios by adding constraints to a model to avoid making large prediction errors.
This protects ML models from poor generalizability by reducing model complexity and minimizing model feature space [@doi:10.1371/journal.pgen.1004754, @doi:10.1002/sim.6782]. (Figure[@fig:3]a)
Examples of ML methods with regularization include ridge regression, LASSO regression, and elastic net regression [@doi:10.1111/j.1467-9868.2005.00503.x], among others.

> Regularization is often used in exploring functional role of variants in rare disease and immune cell signature discovery studies; much like rare disease, these examples need to accommodate sparsity in data.

Author: Justin: Not clear

Author: ```suggestion
Use of regularization is evident in studies examining immune cell signatures, or identifying important disease relevant genes in rare disease, both of which need to accommodate sparsity in data.
```

Author: on second thought, this sentence is now redundant since we are now including an example specifically in rare disease. Previously we had an example that was a non-rare disease, but a sparse dataset. So we needed this segue.

Author: ```suggestion
```

> For example, LASSO regularization reduced the number of features (genes) included as features in an ML model designed to classify brain tissue regions of amyotrophic lateral sclerosis (ALS) patients. [@doi:10.1186/s10020-023-00603-y]

Author: Justin: This doesn’t really demonstrate the purpose of LASSO. Would focus on LASSO as performing feature selection and sparse models to aid in interpretability.

Author: [In the next line] Justin: What is EN doing that might led to better performance, and what about the underlying data is amenable to EN?

Author: ```suggestion
LASSO regularization helped select a few informative genes as features to include in models classifying amyotrophic lateral sclerosis (ALS) patients and healthy patients with high accuracy based on brain tissue gene expression, thus making the models more interpretable. [@doi:10.1186/s10020-023-00603-y]
```

Author: The edit for line 32 is also included in the above comment

## Reducing misinterpretation of model output with statistical techniques

The successful application of ML can be improved by meeting certain conditions.
First, the dataset contains sufficient representation from each class such that relevant variability from that class is captured.
Second, the dataset is complete; all samples have measurements for all variables in the dataset (i.e., the dataset is not “sparse”, it is not missing data for some of the samples).
Third, there is no ambiguity about the labels for the samples in the dataset (i.e., no “label-noise”).

Rare disease datasets violate many of these assumptions.

> There is generally high class imbalance due to small number of samples for specific classes, such as only a few patients with a particular rare disease in a health records dataset.

Reviewer 2: I could go either way on this suggestion:
```suggestion
There is generally high class imbalance due to small number of samples for specific classes, such as only a few patients with a particular rare disease in a health records dataset, which can require special consideration for evaluation (Box 2).
```

> There is generally high class imbalance due to small number of samples for specific classes, such as only a few patients with a particular rare disease in a health records dataset, which can require special consideration for evaluation (Box 2).

Author: ```suggestion
Small number of samples for specific classes fail to fully capture the sample variability in those classes, e.g. only a few patients with a particular rare disease in a health records dataset, which can require special consideration for evaluation (Box 2).
```

Overfit models tend to rely on patterns that are unique to the training data, such as the clinical coding practices at a hospital, and not generalize to new data such as data collected at different hospitals. [@isbn:0262035618; @pmc:PMC8238368]
Regularization approaches can help mitigate these scenarios by adding constraints to a model to avoid making large prediction errors.
This protects ML models from poor generalizability by reducing model complexity and minimizing model feature space [@doi:10.1371/journal.pgen.1004754, @doi:10.1002/sim.6782]. (Figure[@fig:3]a)
Examples of ML methods with regularization include ridge regression, LASSO regression, and elastic net regression [@doi:10.1111/j.1467-9868.2005.00503.x], among others.
Regularization is often used in exploring functional role of variants in rare disease and immune cell signature discovery studies; much like rare disease, these examples need to accommodate sparsity in data.
For example, LASSO regularization reduced the number of features (genes) included as features in an ML model designed to classify brain tissue regions of amyotrophic lateral sclerosis (ALS) patients. [@doi:10.1186/s10020-023-00603-y]

> In the context of rare immune cell signature discovery, where a few genes or features are expected to distinguish between immune cell types, elastic-net regression was able to exclude groups of uninformative genes by reducing their coefficients to zero. [@doi:10.1186/s12859-019-2994-z, @https://doi.org/10.1111/j.1467-9868.2005.00503.x]

Author: ```suggestion
In the context of rare immune cell signature discovery, where a few genes or features are expected to distinguish between immune cell types, elastic-net regression was able to exclude groups of uninformative genes by reducing their contribution to zero. [@doi:10.1186/s12859-019-2994-z, @https://doi.org/10.1111/j.1467-9868.2005.00503.x]
```
